### PR TITLE
Centralize logic for Religious holidays

### PR DESCRIPTION
**Pull request by tinohager**

### CODE REVIEW

File: Src/Nager.Date/PublicHolidays/AlandProvider.cs [hunk 1]
```diff
@@ -26,24 +26,29 @@ public AlandProvider(ICatholicProvider catholicProvider)
...
-            var easterSunday = this._catholicProvider.EasterSunday(year);
 
             var thirdFridayInJune = DateSystem.FindDay(year, Month.June, DayOfWeek.Friday, Occurrence.Third);
             var thirdSaturdayInJune = DateSystem.FindDay(year, Month.June, DayOfWeek.Saturday, Occurrence.Third);
             var firstSaturdayInNovember = DateSystem.FindDay(year, Month.November, DayOfWeek.Saturday, Occurrence.First);
 
+            var easterMonday = this._catholicProvider.EasterMonday("Annandag påsk", year, countryCode);
+            easterMonday.SetLaunchYear(1642);
+
+            var easterSunday = this._catholicProvider.EasterSunday("Easter Sunday", year, countryCode);
```
Comment (samtrion) [inline Src/Nager.Date/PublicHolidays/AlandProvider.cs:37]: Missing Translation - `Påskdagen`

File: Src/Nager.Date/PublicHolidays/AustraliaProvider.cs [hunk 2]
```diff
@@ -66,10 +67,9 @@ public IEnumerable<PublicHoliday> Get(int year)
             items.Add(new PublicHoliday(secondMondayInMarch, "March Public Holiday", "March Public Holiday", countryCode, null, new string[] { "AUS-SA" }));
             items.Add(new PublicHoliday(secondMondayInMarch, "Eight Hours Day", "Eight Hours Day", countryCode, null, new string[] { "AUS-TAS" }));
             items.Add(new PublicHoliday(secondMondayInMarch, "Labour Day", "Labour Day", countryCode, null, new string[] { "AUS-VIC" }));
-            items.Add(new PublicHoliday(easterSunday.AddDays(-2), "Good Friday", "Good Friday", countryCode));
-            items.Add(new PublicHoliday(easterSunday.AddDays(-1), "Easter Eve", "Holy Saturday", countryCode, null, new string[] { "AUS-ACT", "AUS-NSW", "AUS-NT", "AUS-QLD", "AUS-SA", "AUS-VIC" }));
-            items.Add(new PublicHoliday(easterSunday, "Easter Sunday", "Easter Sunday", countryCode, null, new string[] { "AUS-ACT", "AUS-NSW", "AUS-VIC" }));
-            items.Add(new PublicHoliday(easterSunday.AddDays(1), "Easter Monday", "Easter Monday", countryCode));
+            items.Add(this._catholicProvider.GoodFriday("Good Friday", year, countryCode));
+            items.Add(easterSunday);
```
Comment (samtrion) [inline Src/Nager.Date/PublicHolidays/AustraliaProvider.cs:71]: Missing Entry - Easter Eve/Holy Saturday

File: Src/Nager.Date/PublicHolidays/FranceProvider.cs [hunk 3]
```diff
@@ -25,18 +25,22 @@ public FranceProvider(ICatholicProvider catholicProvider)
...
+            easterMonday.SetLaunchYear(1642);
+
+            var goodFriday = this._catholicProvider.GoodFriday("Vendredi saint", year, countryCode);
+            goodFriday.SetCounties("FR-A", "FR-57");
 
             var items = new List<PublicHoliday>();
             items.Add(new PublicHoliday(year, 1, 1, "Jour de l'an", "New Year's Day", countryCode, 1967));
-            items.Add(new PublicHoliday(easterSunday.AddDays(-2), "Vendredi saint", "Good Friday", countryCode, null, new string[] { "FR-A", "FR-57" }));
-            items.Add(new PublicHoliday(easterSunday.AddDays(1), "Lundi de Pâques", "Easter Monday", countryCode, 1642));
+            items.Add(goodFriday);
```
Comment (samtrion) [inline Src/Nager.Date/PublicHolidays/FranceProvider.cs:37]: Missing reference for EasterMonday

File: Src/Nager.Date/PublicHolidays/MontenegroProvider.cs [hunk 4]
```diff
@@ -44,19 +43,31 @@ public IEnumerable<PublicHoliday> Get(int year)
 
             #region Orthodox holidays
 
+            var easterMondayOrthodox = this._orthodoxProvider.EasterMonday("Vaskrs", year, countryCode);
+            easterMondayOrthodox.SetType(PublicHolidayType.Optional);
+
             items.Add(new PublicHoliday(year, 1, 6, "Badnji dan", "Orthodox Christmas Eve", countryCode, null, null, PublicHolidayType.Optional));
             items.Add(new PublicHoliday(year, 1, 7, "Božić", "Orthodox Christmas Day", countryCode, null, null, PublicHolidayType.Optional));
             items.Add(new PublicHoliday(year, 1, 8, "Božić", "Orthodox Christmas Day", countryCode, null, null, PublicHolidayType.Optional));
             items.Add(new PublicHoliday(orthodoxEasterSunday.AddDays(-2), "Vaskrs", "Orthodox Good Friday", countryCode, null, null, PublicHolidayType.Optional));
```
Comment (samtrion) [inline Src/Nager.Date/PublicHolidays/MontenegroProvider.cs:52]: Replace with Orthodox GoodFriday?

Comment (tinohager) [reply Src/Nager.Date/PublicHolidays/MontenegroProvider.cs:52]: https://en.wikipedia.org/wiki/Public_holidays_in_Montenegro

**Pull request merged**